pkg/webhooks/server.go

```diff
@@ -124,6 +125,8 @@ type WebhookServer struct {
 	grController *generate.Controller
 
 	promConfig *metrics.PromConfig
+
+	sync.RWMutex
```

Comment:
Do we really want to embed the mutex ?

Reply:
At least can we embed a pointer instead ?

Reply:
I’m not sure the `is a` relationship is correct 🤔

Reply:
>I’m not sure the is a relationship is correct 

Can you elaborate more on this? What do you mean?

Reply:
I feel like it’s more an `has a` relationship ie the mutex should be a regular field, not an embed.

Reply:
See https://stackoverflow.com/questions/49002/prefer-composition-over-inheritance even if there is no inheritance in golang it still applies to duc typing.

Reply:
Agree with embedding (examples suggest both ways if you check in k8s codebase as well),  but if i remember correctly some golang best practice suggests "If you use a struct by pointer, then the mutex should be a non-pointer field on it"

Reply:
If the mutex is a non-pointer field, better not use the containing struct by non-pointer.
On the other hand, using a mutex pointer makes you safe in all cases.

Reply:
What does go vet says ?

Reply:
`go vet` does not complain